js/main.js

```diff
@@ -981,6 +998,26 @@ require([
 	}
 
 	//Details menu
+	document.getElementById('details_name').onchange = function () {
+		if(menuPoint == undefined)
+			return;
+
+		let val = document.getElementById("details_name").value;
+		menuPoint.array[menuPoint.index].name = val;
+	
+		redrawMap();
+	}
+
+	document.getElementById('details_show_name').onchange = function () {
```

Comment:
Is this really needed as a concept? I think it would be more intuitive to just have the default value as an empty string (and also convert any legacy points that have that value undefined to an empty string when loaded), and if a name is actually set then we just always render that.

Then we can also remove the boolean flag and check `mapObjects.points[i].name === ""` instead.